notebooks/writing_a_trace_transform_cpu_offloading.ipynb

```diff
@@ -0,0 +1,871 @@
+    "\n",
+    "        x = make(shape)\n",
+    "        args, kwargs = (x,), {}\n",
+    "\n",
+    "    return model, args, kwargs, cfg\n",
+    "\n",
+    "def print_memory_usage_and_benchmark(name):\n",
+    "    print(f\"{name} took -\")\n",
+    "    model, args, kwargs, cfg = get_model_and_args()\n",
+    "\n",
+    "    if name == 'thunder':\n",
+    "        jmodel = thunder.jit(model)\n",
+    "    elif name == 'thunder_offload':\n",
+    "        jmodel = thunder.jit(model, transforms=[CPUOffloading()])\n",
+    "    else:\n",
+    "        raise RuntimeError(\"Received invalid value for `name` - try `thunder` or `thunder_offload`.\")\n",
+    "\n",
+    "    memory_after_model_load = torch.cuda.max_memory_allocated() / 1e9\n",
+    "    print(f\"Peak memory after loading the model : {memory_after_model_load} GB\")\n",
+    "\n",
+    "    a = jmodel(*args, **kwargs)\n",
+    "\n",
+    "    memory_after_forward = torch.cuda.max_memory_allocated() / 1e9\n",
+    "    print(f\"Peak memory after forward the model : {memory_after_forward} GB\")\n",
+    "\n",
+    "    g = torch.rand_like(a)\n",
+    "    actual_grads = torch.autograd.grad(a, model.parameters(), g)\n",
+    "\n",
+    "    memory_after_backward = torch.cuda.max_memory_allocated() / 1e9\n",
+    "    print(f\"Peak memory after backward the model : {memory_after_backward} GB\")\n",
+    "\n",
+    "    del a, g, actual_grads  # Clear data which is not required for benchmark to free some memory.\n",
+    "    gc.collect()\n",
+    "    torch.cuda.empty_cache()\n",
+    "\n",
+    "    measurement = benchmark(jmodel, model, args, kwargs)\n",
+    "    print(f\"Benchmark Timings - mean : {measurement.mean} - median {measurement.median}\")\n",
+    "\n",
+    "    del jmodel, model, cfg, args, kwargs\n",
+    "\n",
+    "    clear_memory()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "thunder took -\n",
+      "Peak memory after loading the model : 10.311430656 GB\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[W815 16:01:03.874001433 sdp_utils.cpp:455] Warning: 1Torch was not compiled with flash attention. (function operator())\n",
+      "[W815 16:01:03.874014103 sdp_utils.cpp:504] Warning: 1Torch was not compiled with memory efficient attention. (function operator())\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Peak memory after forward the model : 39.679975424 GB\n",
+      "Peak memory after backward the model : 47.024185344 GB\n",
+      "Benchmark Timings - mean : 3.9535594288026914 - median 3.9535594288026914\n",
+      "CUDA Memory has been cleared and currently allocated memory is  8.192e-06\n"
+     ]
+    }
+   ],
+   "source": [
+    "# print_memory_usage_and_benchmark(\"thunder\")"
```

Comment:
Supercool tutorial! :heart: 

Nit: Maybe we could add some small comment to advise the reader to uncomment:
```suggestion
    "# Uncomment this to run the benchmarks.",
    "# print_memory_usage_and_benchmark(\"thunder\")"
```